### DISCUSSION TITLE
Create many-to-one relationship Prompt Foreign key constraint is incorrectly formed

### DISCUSSION THREAD

**Discussion by exqmjmz** [directus/directus#15739] [Archive]
### Describe the Bug

When creating a many-to-one relationship, select the model, click Save, and an error will be prompted.
Ignore errors, create fields.
The interface shows that no collection is selected.

### To Reproduce

<img width="911" alt="image" src="https://user-images.githubusercontent.com/12406935/192046483-3ae7f47d-8ab3-4cb5-9525-42cf474f183a.png">

<img width="865" alt="image" src="https://user-images.githubusercontent.com/12406935/192046539-feb37000-62a8-4a4c-8f0c-333b2c9822c5.png">

<img width="693" alt="image" src="https://user-images.githubusercontent.com/12406935/192046600-b62b9968-0e82-4849-a8e5-45aebaee847f.png">

<img width="1003" alt="image" src="https://user-images.githubusercontent.com/12406935/192046655-f20180c3-2bd7-4349-9b16-a919f2dfd31d.png">

<img width="1008" alt="image" src="https://user-images.githubusercontent.com/12406935/192046676-4483d906-1ae3-4b4f-ac94-c4855e2b4e71.png">

### Errors Shown

{
  "errors": [
    {
      "message": "alter table `menu_children` add constraint `menu_children_master_foreign` foreign key (`master`) references `menu_master` (`id`) on delete NO ACTION - ER_CANT_CREATE_TABLE: Can't create table `directus`.`menu_children` (errno: 150 \"Foreign key constraint is incorrectly formed\")",
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR"
      }
    }
  ]
}

### What version of Directus are you using?

9.18.1

### What version of Node.js are you using?

16.15

### What database are you using?

mariadb 10.5.17

### What browser are you using?

edge

### How are you deploying Directus?

running locally

**Comment by azrikahar** [comment]
Seems like this might be the same error as #13721. Would you mind trying the solution shared here: https://github.com/directus/directus/discussions/13721#discussioncomment-3317325, to check and see whether differing collations could be the cause here?

**Reply by gokaybiz** [reply]
Hey! 
I am facing the same problem in postgresql.

`"detail": "\"chapter_name\" and \"id\" key columns have different type: character varying and integer.",`
`"message": "alter table \"chapters\" add constraint \"chapters_chapter_name_foreign\" foreign key (\"chapter_name\") references \"courses\" (\"id\") on delete NO ACTION - Cannot create \"chapters_chapter_name_foreign\" foreign key constraint"`
but the ui won't let me choose id
